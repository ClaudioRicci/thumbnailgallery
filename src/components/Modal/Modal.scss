button {
  display: block;
  margin: 2rem auto;
  font-size: 1.2rem;
  padding: 1rem;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
}

.images {
  img {
    width: 100%;
  }
}
.modal-overlay {
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.85);
  display: flex;
  padding: 1em;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 4;
}

.modal {
  background: #fff;
  border-radius: 1em;
  padding: 0 1rem 1rem;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 80%;
  height: auto;

  .card {
    max-width: 100%;
    height: 638px;

    img {
      float: left;
      width: 60%;
      margin: 1em 1.4em;
      position: relative;
      z-index: 2;
      max-height: 550px;
      max-width: 814px;
    }

    .detailsSurround + .button {
      width: auto;
      position: absolute;
      bottom: 20px;
      z-index: 1;
      right: 2%;
    }

    .leftSectionSurround {
      left: 43%;
      width: 45%;
      border-top: 120px solid #fff;
      border-left: 100px solid transparent;
      border-bottom: none;
      border-right: none;
      top: 60px;
      position: relative;
      z-index: 1;
      .leftSection {
        top: -108px;
        left: 50px;
      }
    }

    .rightSectionSurround {
      right: -69.6%;
      width: 28%;
      top: -50px;
      font-size: 1.5em;
      border-bottom: 150px solid #000;
    }

    .moreInfoButton {
      display: none;
    }

    .carwowRating {
      top: 110px;
      position: absolute;
      left: 4%;
    }

    .summary {
      display: inline-flex;
      position: absolute;
      height: 0;
      border-bottom: 170px solid #fff;
      border-left: 100px solid transparent;
      z-index: 1;
      bottom: 15px;
      width: fit-content;
      left: 0;
      p {
        line-height: 1.6em;
        font-size: 1.6em;
        text-align: left;
        padding: 0.4em 1em 0 0;
      }
    }
  }

  .card:hover {
    box-shadow: none;
  }
}
