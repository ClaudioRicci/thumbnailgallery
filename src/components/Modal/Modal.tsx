import React, { useLayoutEffect } from "react";
import Frame from "../Frame/Frame";
import { pure } from "recompose";
import "./Modal.scss";

const Modal: React.SFC = props => {
  function shutModal() {
    const modalOverlay: any = document.getElementById("modal-overlay");
    modalOverlay.style.display = "none";
  }

  function useLockBodyScroll() {
    useLayoutEffect(() => {
      const originalStyle = window.getComputedStyle(document.body).overflow;
      document.body.style.overflow = "hidden";
      return () => (document.body.style.overflow = originalStyle);
    }, []);
  }

  useLockBodyScroll();

  return (
    <div id="modal-overlay" className="modal-overlay" data-testid="modal">
      <div className="modal">
        <Frame {...props} />
        <button className="button closeButton" onClick={() => shutModal()}>
          <span>Close</span>
        </button>
      </div>
    </div>
  );
};

export default pure(Modal);
