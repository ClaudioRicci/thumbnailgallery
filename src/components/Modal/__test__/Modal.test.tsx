import React from "react";
import ReactDOM from "react-dom";
import Modal from "../Modal";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Modal></Modal>, div);
});

test("Renders Modal Correctly", done => {
  const { getByTestId } = render(<Modal />);
  expect(getByTestId("modal")).toHaveTextContent("Get Offers");
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Modal></Modal>).toJSON();
  expect(tree).toMatchSnapshot();
});
