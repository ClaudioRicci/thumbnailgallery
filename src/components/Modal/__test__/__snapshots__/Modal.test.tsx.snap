// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="modal-overlay"
  data-testid="modal"
  id="modal-overlay"
>
  <div
    className="modal-overlay__modal"
  >
    <article
      className="frame"
      data-testid="frame"
    >
      <img
        alt="Image is copyright of CCO, sourced by unknown."
        src="https://splashbase.s3.amazonaws.com/unsplash/large/14ZGHd9"
      />
    </article>
    <button
      className="Close"
      data-testid="button"
    >
      <span>
        Close
      </span>
    </button>
  </div>
</div>
`;
